modeltranslation/admin.py

```diff
@@ -34,8 +34,13 @@ def _get_declared_fieldsets(self, request, obj=None):
         # Take custom modelform fields option into account
         if not self.fields and hasattr(self.form, '_meta') and self.form._meta.fields:
             self.fields = self.form._meta.fields
-        if self.fieldsets:
-            return self._patch_fieldsets(self.fieldsets)
+        fieldsets = (
```

Comment:
Maybe we can use get_fieldsets here?

It would be better, because `add_fieldsets` is not a standard ModelAdmin attribute. UserAdmin overrides get_fieldsets to use this attribute when creating a user. (source: https://docs.djangoproject.com/en/4.1/topics/auth/customizing/)

Reply:
We can't use `get_fieldsets` here because `_get_declared_fieldsets` (where these changes are made) is called by `get_fieldsets` in `TranslationAdmin` class. I suppose we could move it to `get_fieldsets` and pass it as argument to `_get_fieldsets_pre_form_or_formset` and `_get_declared_fieldsets`?.

Reply:
Hmm, no, let's use your original code then. I think it's fine.

But, please add a comment about `add_fieldsets` being non-standard attribute.

Reply:
sure, will do